src/app/modules/main/app_search/module.nim

```diff
@@ -110,7 +110,8 @@ method prepareLocationMenuModel*(self: Module) =
   for c in communities:
     items.add(self.buildLocationMenuForCommunity(c))
 
-  self.view.locationMenuModel().setItems(items)
+  if (items.len != self.view.locationMenuModel().len):
+    self.view.locationMenuModel().setItems(items)
```

Comment:
I had to add this condition because calling `setItems` repeatetly while inside a community caused a crash. `beginResetModel` just failed for no reason sometimes, but reproducibly.

I'm still unsure why, so this condition above fixes it, because it seems like it was the repeated reset of the model that caused issues.

@osmaczko since you're the one that created the code here, can look at it and let me know if you think it might cause issues? I tested it it out and it seemed to work well, even if I add new channels, the search seemed to still work.

Reply:
@jrainville this check `if (items.len != self.view.locationMenuModel().len):` as I see will introduce new issue, cause every time one runs "app search popup" we're building a location menu based on the current state. With this check if some community/chat change image/color or new the length will be the same, but the content won't be accurate.

A crash you mentioned is most likely caused on the qml side, cause that popup menu is being built from the model and some submenus/actions are added dynamically, maybe we should search in that direction for the crash reason. Maybe we should also experiment a bit with signals sent when we're setting a new set of items.

Reply:
Hm..:question: I am not the author of this code :thinking:

Reply:
@saledjenic alright I finally figured out the issue. See AppSearch.qml. The problem was that the old list was not emptied correctly , leaving some remains, so if you had multiple communities, there ended up being a wrong memory access. Also, the duplicated community, when hovered crashed because again, there was no model for it.

@osmaczko my bad, I had forgotten that it was Sale who created it.